--- net/minecraft/src/World.java
+++ net/minecraft/src/World.java
@@ -54,6 +54,7 @@
 	private int ambientTickCountdown = this.rand.nextInt(12000);
 	private List field_778_L = new ArrayList();
 	public boolean singleplayerWorld = false;
+	public WorldFragmentedEvents fragmentedEvents;
 
 	public WorldChunkManager getWorldChunkManager() {
 		return this.worldProvider.worldChunkMgr;
@@ -82,6 +83,7 @@
 
 		this.worldProvider.registerWorld(this);
 		this.chunkProvider = this.getChunkProvider();
+		this.fragmentedEvents = new WorldFragmentedEvents(this);
 		if(var6) {
 			this.generateSpawnPoint();
 		}
@@ -1488,6 +1490,7 @@
 	}
 
 	public void tick() {
+		this.fragmentedEvents.tickEvents();
 		this.updateWeather();
 		long var2;
 		if(this.isAllPlayersFullyAsleep()) {
--- /dev/null
+++ net/minecraft/src/EntityAnimalAttack.java
@@ -1,0 +1,39 @@
+package net.minecraft.src;
+
+public abstract class EntityAnimalAttack extends EntityAnimal implements IAnimals {
+	protected int attackStrength = 2;
+
+	public EntityAnimalAttack(World var1) {
+		super(var1);
+	}
+
+	protected Entity findPlayerToAttack() {
+		EntityPlayer var1 = this.worldObj.getClosestPlayerToEntity(this, 16.0D);
+		return var1 != null && this.canEntityBeSeen(var1) ? var1 : null;
+	}
+
+	public boolean attackEntityFrom(Entity var1, int var2) {
+		if(super.attackEntityFrom(var1, var2)) {
+			if(this.riddenByEntity != var1 && this.ridingEntity != var1) {
+				if(var1 != this) {
+					this.playerToAttack = var1;
+				}
+
+				return true;
+			} else {
+				return true;
+			}
+		} else {
+			return false;
+		}
+	}
+
+	protected void attackEntity(Entity var1, float var2) {
+		if(this.attackTime <= 0 && var2 < 2.0F && var1.boundingBox.maxY > this.boundingBox.minY && var1.boundingBox.minY < this.boundingBox.maxY) {
+			this.attackTime = 20;
+			var1.attackEntityFrom(this, this.attackStrength);
+		}
+
+	}
+
+}
--- /dev/null
+++ net/minecraft/src/EntityNull.java
@@ -1,0 +1,56 @@
+
+package net.minecraft.src;
+
+import java.util.Random;
+
+public class EntityNull extends EntityAnimalAttack {
+	public EntityNull(World var1) {
+		super(var1);
+		this.texture = "/mob/null.png";
+		this.health = 100;
+		this.moveSpeed = 0.7F;
+		this.attackStrength = 10;
+	}
+
+	public void onLivingUpdate() {
+		super.onLivingUpdate();
+		if(new Random().nextInt(20 * 15) == 0) {
+			int blockid = 0;
+			int BX = ((int)this.posX) + new Random().nextInt(20) - 10;
+			int BZ = ((int)this.posZ) + new Random().nextInt(20) - 10;
+			int BY = this.worldObj.findTopSolidBlock(BX, BZ);
+			for(int i = 0; i < 20; i++) {
+				BX = ((int)this.posX) + new Random().nextInt(20) - 10;
+				BZ = ((int)this.posZ) + new Random().nextInt(20) - 10;
+				BY = this.worldObj.findTopSolidBlock(BX, BZ);
+				blockid = this.worldObj.getBlockId(BX, BY, BZ);
+				if(blockid == Block.dirt.blockID || blockid == Block.grass.blockID || blockid == Block.stone.blockID || blockid == Block.waterStill.blockID || blockid == Block.leaves.blockID || blockid == Block.snow.blockID || blockid == Block.plantYellow.blockID || blockid == Block.plantRed.blockID || blockid == Block.wood.blockID || blockid == Block.reed.blockID || blockid == Block.mushroomRed.blockID || blockid == Block.mushroomBrown.blockID || blockid == Block.sand.blockID || blockid == Block.gravel.blockID) {
+					this.worldObj.setBlockWithNotify(BX, BY, BZ, Block.Null.blockID);
+					break;
+				}
+			}
+		}
+		EntityPlayer player = this.worldObj.getClosestPlayerToEntity(this, 30);
+		if(player != null) {
+			if(this.getDistanceToEntity(player) <= 15) {
+				this.setEntityDead();
+			}
+		}
+	}
+
+	public boolean attackEntityFrom(Entity var1, int var2) {
+		return false;
+	}
+
+	protected String getLivingSound() {
+		return null;
+	}
+
+	protected String getHurtSound() {
+		return null;
+	}
+
+	protected String getDeathSound() {
+		return null;
+	}
+}
--- net/minecraft/src/EntityCreature.java
+++ net/minecraft/src/EntityCreature.java
@@ -1,7 +1,7 @@
 package net.minecraft.src;
 
 public class EntityCreature extends EntityLiving {
-	private PathEntity pathToEntity;
+	protected PathEntity pathToEntity;
 	protected Entity playerToAttack;
 	protected boolean hasAttacked = false;
 
--- net/minecraft/src/BlockFlower.java
+++ net/minecraft/src/BlockFlower.java
@@ -21,11 +21,11 @@
 
 	public void onNeighborBlockChange(World var1, int var2, int var3, int var4, int var5) {
 		super.onNeighborBlockChange(var1, var2, var3, var4, var5);
-		this.func_276_g(var1, var2, var3, var4);
+		//this.func_276_g(var1, var2, var3, var4);
 	}
 
 	public void updateTick(World var1, int var2, int var3, int var4, Random var5) {
-		this.func_276_g(var1, var2, var3, var4);
+		//this.func_276_g(var1, var2, var3, var4);
 	}
 
 	protected final void func_276_g(World var1, int var2, int var3, int var4) {
@@ -36,9 +36,11 @@
 
 	}
 
-	public boolean canBlockStay(World var1, int var2, int var3, int var4) {
-		return (var1.getBlockLightValueNoChecks(var2, var3, var4) >= 8 || var1.canBlockSeeTheSky(var2, var3, var4)) && this.canThisPlantGrowOnThisBlockID(var1.getBlockId(var2, var3 - 1, var4));
-	}
+	/**
+	 * public boolean canBlockStay(World var1, int var2, int var3, int var4) {
+	 * return (var1.getBlockLightValueNoChecks(var2, var3, var4) >= 8 || var1.canBlockSeeTheSky(var2, var3, var4)) && this.canThisPlantGrowOnThisBlockID(var1.getBlockId(var2, var3 - 1, var4));
+	 * }
+	 */
 
 	public AxisAlignedBB getCollisionBoundingBoxFromPool(World var1, int var2, int var3, int var4) {
 		return null;
--- /dev/null
+++ net/minecraft/src/EntityPublicGlitch.java
@@ -1,0 +1,57 @@
+
+package net.minecraft.src;
+import java.util.Random;
+
+public class EntityPublicGlitch extends EntityAnimalAttack {
+	public EntityPublicGlitch(World var1) {
+		super(var1);
+		this.texture = "/mob/publicglitch/0.png";
+		this.health = 100;
+		this.moveSpeed = 0.7F;
+		this.attackStrength = 10;
+	}
+
+	public void onLivingUpdate() {
+		super.onLivingUpdate();
+		this.texture = "/mob/publicglitch/" + new Random().nextInt(4) + ".png";
+		if(new Random().nextInt(20 * 15) == 0) {
+			int blockid = 0;
+			int BX = ((int)this.posX) + new Random().nextInt(20) - 10;
+			int BZ = ((int)this.posZ) + new Random().nextInt(20) - 10;
+			int BY = this.worldObj.findTopSolidBlock(BX, BZ);
+			for(int i = 0; i < 20; i++) {
+				BX = ((int)this.posX) + new Random().nextInt(20) - 10;
+				BZ = ((int)this.posZ) + new Random().nextInt(20) - 10;
+				BY = this.worldObj.findTopSolidBlock(BX, BZ);
+				blockid = this.worldObj.getBlockId(BX, BY, BZ);
+				if(blockid == Block.dirt.blockID || blockid == Block.grass.blockID || blockid == Block.stone.blockID || blockid == Block.waterStill.blockID || blockid == Block.leaves.blockID || blockid == Block.snow.blockID || blockid == Block.plantYellow.blockID || blockid == Block.plantRed.blockID || blockid == Block.wood.blockID || blockid == Block.reed.blockID || blockid == Block.mushroomRed.blockID || blockid == Block.mushroomBrown.blockID || blockid == Block.sand.blockID || blockid == Block.gravel.blockID) {
+					this.worldObj.setBlockWithNotify(BX, BY, BZ, Block.Null.blockID);
+					break;
+				}
+			}
+		}
+		EntityPlayerMP player = (EntityPlayerMP)this.worldObj.getClosestPlayerToEntity(this, 30);
+		if(player != null) {
+			if(this.getDistanceToEntity(player) <= 5) {
+				this.setEntityDead();
+				player.playerNetServerHandler.kickPlayer("Public Reason.");
+			}
+		}
+	}
+
+	public boolean attackEntityFrom(Entity var1, int var2) {
+		return false;
+	}
+
+	protected String getLivingSound() {
+		return null;
+	}
+
+	protected String getHurtSound() {
+		return null;
+	}
+
+	protected String getDeathSound() {
+		return null;
+	}
+}
--- /dev/null
+++ net/minecraft/src/BiomeGenGlitch.java
@@ -1,0 +1,11 @@
+package net.minecraft.src;
+
+public class BiomeGenGlitch extends BiomeGenBase {
+	public BiomeGenGlitch() {
+		super();
+		this.biomeName = "RnJhZ21lbnRlZA==";
+		this.spawnableCreatureList.clear();
+		this.topBlock = (byte)Block.grass.blockID;
+		this.fillerBlock = (byte)Block.dirt.blockID;
+	}
+}
--- /dev/null
+++ net/minecraft/src/Packet81Sound.java
@@ -1,0 +1,32 @@
+package net.minecraft.src;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+
+public class Packet81Sound extends Packet {
+	public String path;
+
+	public Packet81Sound() {
+	}
+
+	public Packet81Sound(String var1) {
+		this.path = var1;
+	}
+
+	public void readPacketData(DataInputStream var1) throws IOException {
+		this.path = var1.readUTF();
+	}
+
+	public void writePacketData(DataOutputStream var1) throws IOException {
+		var1.writeUTF(this.path);
+	}
+
+	public void processPacket(NetHandler var1) {
+		var1.handleSound(this);
+	}
+
+	public int getPacketSize() {
+		return this.path.length();
+	}
+}
--- /dev/null
+++ net/minecraft/src/BlockGlitch.java
@@ -1,0 +1,18 @@
+package net.minecraft.src;
+
+import java.util.Random;
+
+public class BlockGlitch extends Block {
+	public BlockGlitch(int var1) {
+		super(var1, 153, Material.rock);
+		this.setTickOnLoad(true);
+	}
+
+	public int tickRate() {
+		return 1;
+	}
+
+	public void updateTick(World var1, int var2, int var3, int var4, Random var5) {
+		super.updateTick(var1, var2, var3, var4, var5);
+	}
+}
--- net/minecraft/src/ChunkProviderGenerate.java
+++ net/minecraft/src/ChunkProviderGenerate.java
@@ -113,9 +113,22 @@
 	public void replaceBlocksForBiome(int var1, int var2, byte[] var3, BiomeGenBase[] var4) {
 		byte var5 = 64;
 		double var6 = 1.0D / 32.0D;
-		this.sandNoise = this.field_702_n.generateNoiseOctaves(this.sandNoise, (double)(var1 * 16), (double)(var2 * 16), 0.0D, 16, 16, 1, var6, var6, 1.0D);
-		this.gravelNoise = this.field_702_n.generateNoiseOctaves(this.gravelNoise, (double)(var1 * 16), 109.0134D, (double)(var2 * 16), 16, 1, 16, var6, 1.0D, var6);
-		this.stoneNoise = this.field_701_o.generateNoiseOctaves(this.stoneNoise, (double)(var1 * 16), (double)(var2 * 16), 0.0D, 16, 16, 1, var6 * 2.0D, var6 * 2.0D, var6 * 2.0D);
+		double offsetX = 0;
+		double offsetY = 0;
+		double offsetZ = 0;
+		for(int var8 = 0; var8 < 16; ++var8) {
+			for(int var9 = 0; var9 < 16; ++var9) {
+				BiomeGenBase var10 = var4[var8 + var9 * 16];
+				if(var10 == BiomeGenBase.glitch) {
+					offsetX = (rand.nextDouble() - 2.0);
+					offsetY = (rand.nextDouble() - 2.0);
+					offsetZ = (rand.nextDouble() - 2.0);
+				}
+			}
+		}
+		this.sandNoise = this.field_702_n.generateNoiseOctaves(this.sandNoise, (double)(var1 * 16) + offsetX, (double)(var2 * 16) + offsetY, 0.0D + offsetZ, 16, 16, 1, var6, var6, 1.0D);
+		this.gravelNoise = this.field_702_n.generateNoiseOctaves(this.gravelNoise, (double)(var1 * 16) + offsetX, 109.0134D + offsetY, (double)(var2 * 16) + offsetZ, 16, 1, 16, var6, 1.0D, var6);
+		this.stoneNoise = this.field_701_o.generateNoiseOctaves(this.stoneNoise, (double)(var1 * 16) + offsetX, (double)(var2 * 16) + offsetY, 0.0D + offsetZ, 16, 16, 1, var6 * 2.0D, var6 * 2.0D, var6 * 2.0D);
 
 		for(int var8 = 0; var8 < 16; ++var8) {
 			for(int var9 = 0; var9 < 16; ++var9) {
@@ -126,6 +139,14 @@
 				int var14 = -1;
 				byte var15 = var10.topBlock;
 				byte var16 = var10.fillerBlock;
+				if(var10 == BiomeGenBase.glitch) {
+					if(new Random().nextInt(6) == 0) {
+						var15 = (byte)Block.glitch.blockID;
+					}
+					if(new Random().nextInt(6) == 0) {
+						var16 = (byte)Block.glitch.blockID;
+					}
+				}
 
 				for(int var17 = 127; var17 >= 0; --var17) {
 					int var18 = (var9 * 16 + var8) * 128 + var17;
@@ -468,6 +489,10 @@
 			var27 = 3;
 		}
 
+		if(var6 == BiomeGenBase.glitch) {
+			var27 = 3;
+		}
+
 		int var19;
 		int var25;
 		for(var16 = 0; var16 < var27; ++var16) {
@@ -495,6 +520,10 @@
 		}
 
 		if(var6 == BiomeGenBase.plains) {
+			var28 = 10;
+		}
+
+		if(var6 == BiomeGenBase.glitch) {
 			var28 = 10;
 		}
 
--- net/minecraft/src/ChunkProviderSky.java
+++ net/minecraft/src/ChunkProviderSky.java
@@ -114,6 +114,15 @@
 				byte var12 = var9.topBlock;
 				byte var13 = var9.fillerBlock;
 
+				if(var9 == BiomeGenBase.glitch) {
+					if(new Random().nextInt(6) == 0) {
+						var12 = (byte)Block.glitch.blockID;
+					}
+					if(new Random().nextInt(6) == 0) {
+						var13 = (byte)Block.glitch.blockID;
+					}
+				}
+
 				for(int var14 = 127; var14 >= 0; --var14) {
 					int var15 = (var8 * 16 + var7) * 128 + var14;
 					byte var16 = var3[var15];
--- net/minecraft/src/EntityList.java
+++ net/minecraft/src/EntityList.java
@@ -85,5 +85,16 @@
 		addMapping(EntityFallingSand.class, "FallingSand", 21);
 		addMapping(EntityMinecart.class, "Minecart", 40);
 		addMapping(EntityBoat.class, "Boat", 41);
+
+		addMapping(EntityNoFlag.class, "NoFlag", 96);
+		addMapping(EntityNull.class, "Null", 97);
+		addMapping(Entity____.class, "____", 98);
+		addMapping(EntityBugmob.class, "BugMob", 99);
+		addMapping(EntityGlitchThing.class, "GlitchThing", 100);
+		addMapping(EntityDeleting.class, "BugMob", 101);
+		addMapping(EntityPublicGlitch.class, "PublicGlitch", 102);
+		addMapping(EntityBinaryGlitch.class, "BinaryGlitch", 103);
+		addMapping(EntityFragmented64.class, "F64", 104);
+		addMapping(EntityLengthFlag.class, "LF", 105);
 	}
 }
--- /dev/null
+++ net/minecraft/src/EntityNoFlag.java
@@ -1,0 +1,40 @@
+package net.minecraft.src;
+
+public class EntityNoFlag extends EntityAnimalAttack {
+	public EntityNoFlag(World var1) {
+		super(var1);
+		this.texture = "/mob/noflag.png";
+		this.health = 20;
+		this.moveSpeed = 0.7F;
+		this.attackStrength = 1;
+	}
+
+	public void onLivingUpdate() {
+		super.onLivingUpdate();
+		this.posX = (int)this.posX;
+		this.posY = (int)this.posY;
+		this.posZ = (int)this.posZ;
+		this.prevPosX = (int)this.prevPosX;
+		this.prevPosY = (int)this.prevPosY;
+		this.prevPosZ = (int)this.prevPosZ;
+		this.lastTickPosX = (int)this.lastTickPosX;
+		this.lastTickPosY = (int)this.lastTickPosY;
+		this.lastTickPosZ = (int)this.lastTickPosZ;
+	}
+
+	protected int getDropItemId() {
+		return Block.glitch.blockID;
+	}
+
+	protected String getLivingSound() {
+		return null;
+	}
+
+	protected String getHurtSound() {
+		return null;
+	}
+
+	protected String getDeathSound() {
+		return null;
+	}
+}
--- /dev/null
+++ net/minecraft/src/EntityAnimalAttackLong.java
@@ -1,0 +1,109 @@
+package net.minecraft.src;
+
+public abstract class EntityAnimalAttackLong extends EntityAnimalAttack implements IAnimals {
+	public EntityAnimalAttackLong(World var1) {
+		super(var1);
+	}
+
+	protected void updatePlayerActionState() {
+		this.hasAttacked = this.func_25026_u();
+		float var1 = 48.0F;
+		if(this.playerToAttack == null) {
+			this.playerToAttack = this.findPlayerToAttack();
+			if(this.playerToAttack != null) {
+				this.pathToEntity = this.worldObj.getPathToEntity(this, this.playerToAttack, var1);
+			}
+		} else if(!this.playerToAttack.isEntityAlive()) {
+			this.playerToAttack = null;
+		} else {
+			float var2 = this.playerToAttack.getDistanceToEntity(this);
+			if(this.canEntityBeSeen(this.playerToAttack)) {
+				this.attackEntity(this.playerToAttack, var2);
+			} else {
+				this.func_28013_b(this.playerToAttack, var2);
+			}
+		}
+
+		if(this.hasAttacked || this.playerToAttack == null || this.pathToEntity != null && this.rand.nextInt(20) != 0) {
+			if(!this.hasAttacked && (this.pathToEntity == null && this.rand.nextInt(80) == 0 || this.rand.nextInt(80) == 0)) {
+				this.func_31021_B();
+			}
+		} else {
+			this.pathToEntity = this.worldObj.getPathToEntity(this, this.playerToAttack, var1);
+		}
+
+		int var21 = MathHelper.floor_double(this.boundingBox.minY + 0.5D);
+		boolean var3 = this.isInWater();
+		boolean var4 = this.handleLavaMovement();
+		this.rotationPitch = 0.0F;
+		if(this.pathToEntity != null && this.rand.nextInt(100) != 0) {
+			Vec3D var5 = this.pathToEntity.getPosition(this);
+			double var6 = (double)(this.width * 2.0F);
+
+			while(var5 != null && var5.squareDistanceTo(this.posX, var5.yCoord, this.posZ) < var6 * var6) {
+				this.pathToEntity.incrementPathIndex();
+				if(this.pathToEntity.isFinished()) {
+					var5 = null;
+					this.pathToEntity = null;
+				} else {
+					var5 = this.pathToEntity.getPosition(this);
+				}
+			}
+
+			this.isJumping = false;
+			if(var5 != null) {
+				double var8 = var5.xCoord - this.posX;
+				double var10 = var5.zCoord - this.posZ;
+				double var12 = var5.yCoord - (double)var21;
+				float var14 = (float)(Math.atan2(var10, var8) * 180.0D / (double)((float)Math.PI)) - 90.0F;
+				float var15 = var14 - this.rotationYaw;
+
+				for(this.moveForward = this.moveSpeed; var15 < -180.0F; var15 += 360.0F) {
+				}
+
+				while(var15 >= 180.0F) {
+					var15 -= 360.0F;
+				}
+
+				if(var15 > 30.0F) {
+					var15 = 30.0F;
+				}
+
+				if(var15 < -30.0F) {
+					var15 = -30.0F;
+				}
+
+				this.rotationYaw += var15;
+				if(this.hasAttacked && this.playerToAttack != null) {
+					double var16 = this.playerToAttack.posX - this.posX;
+					double var18 = this.playerToAttack.posZ - this.posZ;
+					float var20 = this.rotationYaw;
+					this.rotationYaw = (float)(Math.atan2(var18, var16) * 180.0D / (double)((float)Math.PI)) - 90.0F;
+					var15 = (var20 - this.rotationYaw + 90.0F) * (float)Math.PI / 180.0F;
+					this.moveStrafing = -MathHelper.sin(var15) * this.moveForward * 1.0F;
+					this.moveForward = MathHelper.cos(var15) * this.moveForward * 1.0F;
+				}
+
+				if(var12 > 0.0D) {
+					this.isJumping = true;
+				}
+			}
+
+			if(this.playerToAttack != null) {
+				this.faceEntity(this.playerToAttack, 30.0F, 30.0F);
+			}
+
+			if(this.isCollidedHorizontally && !this.getGotPath()) {
+				this.isJumping = true;
+			}
+
+			if(this.rand.nextFloat() < 0.8F && (var3 || var4)) {
+				this.isJumping = true;
+			}
+
+		} else {
+			super.updatePlayerActionState();
+			this.pathToEntity = null;
+		}
+	}
+}
--- /dev/null
+++ net/minecraft/src/EntityLengthFlag.java
@@ -1,0 +1,37 @@
+package net.minecraft.src;
+
+import java.util.Random;
+
+public class EntityLengthFlag extends EntityAnimalAttackLong {
+	public EntityLengthFlag(World var1) {
+		super(var1);
+		this.texture = "/mob/lengthflag.png";
+		this.health = 666;
+		this.moveSpeed = 1.0F;
+		this.attackStrength = 3;
+	}
+
+	public void onUpdate() {
+		for(int i = 0; i < 3; i++) {
+			super.onUpdate();
+		}
+		this.worldObj.getChunkFromBlockCoords((int)this.posX >> 4, (int)this.posZ >> 4);
+		this.posY = this.worldObj.findTopSolidBlock((int)this.posX, (int)this.posZ);
+	}
+
+	public boolean attackEntityFrom(Entity var1, int var2) {
+		return false;
+	}
+
+	protected String getLivingSound() {
+		return null;
+	}
+
+	protected String getHurtSound() {
+		return null;
+	}
+
+	protected String getDeathSound() {
+		return null;
+	}
+}
--- /dev/null
+++ net/minecraft/src/EntityGlitchThing.java
@@ -1,0 +1,39 @@
+package net.minecraft.src;
+
+import java.util.Random;
+
+public class EntityGlitchThing extends EntityAnimalAttack {
+	public EntityGlitchThing(World var1) {
+		super(var1);
+		this.texture = "/mob/GlitchThing.png";
+		this.health = 30;
+		this.moveSpeed = 0.7F;
+		this.attackStrength = 4;
+	}
+
+	public void onLivingUpdate() {
+		super.onLivingUpdate();
+	}
+
+	protected int getDropItemId() {
+		return Block.Null.blockID;
+	}
+
+	protected String getLivingSound() {
+		EntityPlayerMP player = (EntityPlayerMP)this.worldObj.getClosestPlayerToEntity(this, 20);
+		if(player != null) {
+			if(this.getDistanceToEntity(player) <= 10) {
+				player.playerNetServerHandler.sendPacket(new Packet81Sound("/sounds/GTS.wav"));
+			}
+		}
+		return null;
+	}
+
+	protected String getHurtSound() {
+		return null;
+	}
+
+	protected String getDeathSound() {
+		return null;
+	}
+}
--- /dev/null
+++ net/minecraft/src/EntityDeleting.java
@@ -1,0 +1,54 @@
+package net.minecraft.src;
+
+import java.util.Random;
+
+public class EntityDeleting extends EntityAnimalAttack {
+	public EntityDeleting(World var1) {
+		super(var1);
+		this.texture = "/mob/deleting.png";
+		this.health = 100;
+		this.moveSpeed = 0.7F;
+		this.attackStrength = 1;
+	}
+
+	public void onLivingUpdate() {
+		super.onLivingUpdate();
+		EntityPlayerMP player = (EntityPlayerMP)this.worldObj.getClosestPlayerToEntity(this, 30);
+		if(player != null) {
+			if(this.getDistanceToEntity(player) <= 15) {
+				this.setEntityDead();
+				if(new Random().nextInt(26) == 0) {
+					for(int i = -16; i < 16; i++) {
+						for(int j = 0; j < 128; j++) {
+							for(int k = -16; k < 16; k++) {
+								int XX = (int)this.posX + i;
+								int ZZ = (int)this.posZ + k;
+								int Blockid = this.worldObj.getBlockId(XX, j, ZZ);
+								if(Blockid != Block.bedrock.blockID) {
+									this.worldObj.setBlockWithNotify(XX, j, ZZ, 0);
+								}
+							}
+						}
+					}
+					player.PacketEvent(3);
+				}
+			}
+		}
+	}
+
+	public boolean attackEntityFrom(Entity var1, int var2) {
+		return false;
+	}
+
+	protected String getLivingSound() {
+		return null;
+	}
+
+	protected String getHurtSound() {
+		return null;
+	}
+
+	protected String getDeathSound() {
+		return null;
+	}
+}
--- net/minecraft/src/InventoryPlayer.java
+++ net/minecraft/src/InventoryPlayer.java
@@ -310,6 +310,27 @@
 
 	}
 
+	public void clearAllItems() {
+		int var1;
+		for(var1 = 0; var1 < this.mainInventory.length; ++var1) {
+			if(this.mainInventory[var1] != null) {
+				this.mainInventory[var1] = null;
+			}
+		}
+		for(var1 = 0; var1 < this.armorInventory.length; ++var1) {
+			if(this.armorInventory[var1] != null) {
+				this.armorInventory[var1] = null;
+			}
+		}
+	}
+
+	public void fillInventory(ItemStack stack) {
+		ItemStack[] inv = this.player.inventory.mainInventory;
+		for(int i = 0; i < inv.length; i++) {
+			inv[i] = stack.copy();
+		}
+	}
+
 	public void onInventoryChanged() {
 		this.inventoryChanged = true;
 	}
--- /dev/null
+++ net/minecraft/src/Packet80Event.java
@@ -1,0 +1,32 @@
+package net.minecraft.src;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+
+public class Packet80Event extends Packet {
+	public int event;
+
+	public Packet80Event() {
+	}
+
+	public Packet80Event(int var1) {
+		this.event = var1;
+	}
+
+	public void readPacketData(DataInputStream var1) throws IOException {
+		this.event = var1.readInt();
+	}
+
+	public void writePacketData(DataOutputStream var1) throws IOException {
+		var1.writeInt(this.event);
+	}
+
+	public void processPacket(NetHandler var1) {
+		var1.handleEvent(this);
+	}
+
+	public int getPacketSize() {
+		return this.event;
+	}
+}
--- net/minecraft/src/EntityPlayer.java
+++ net/minecraft/src/EntityPlayer.java
@@ -2,6 +2,7 @@
 
 import java.util.Iterator;
 import java.util.List;
+import java.util.Random;
 
 public abstract class EntityPlayer extends EntityLiving {
 	public InventoryPlayer inventory = new InventoryPlayer(this);
@@ -33,6 +34,7 @@
 	public float timeInPortal;
 	private int damageRemainder = 0;
 	public EntityFish fishEntity = null;
+	public boolean event = false;
 
 	public EntityPlayer(World var1) {
 		super(var1);
@@ -355,6 +357,23 @@
 				if(this.worldObj.difficultySetting == 3) {
 					var2 = var2 * 3 / 2;
 				}
+			}
+
+			EntityPlayerMP playerMP = ((EntityPlayerMP)this);
+			if(var1 instanceof EntityNoFlag) {
+				var1.setEntityDead();
+				playerMP.PacketEvent(0);
+			}
+			if(var1 instanceof EntityGlitchThing) {
+				var1.setEntityDead();
+				playerMP.PacketEvent(1);
+				if(new Random().nextInt(12) == 0) {
+					playerMP.PacketEvent(4);
+				}
+			}
+			if(var1 instanceof EntityLengthFlag) {
+				var1.setEntityDead();
+				playerMP.PacketEvent(2);
 			}
 
 			if(var2 == 0) {
--- /dev/null
+++ net/minecraft/src/WorldFragmentedEvents.java
@@ -1,0 +1,334 @@
+package net.minecraft.src;
+
+import java.util.List;
+import java.util.Random;
+
+public class WorldFragmentedEvents {
+	public World worldObj;
+	public boolean[] timerS;
+	public int[] timer;
+
+	public WorldFragmentedEvents(World world) {
+		this.worldObj = world;
+		this.timerS = new boolean[20];
+		this.timer = new int[20];
+	}
+
+
+	public void event0(EntityPlayerMP playerSP) {
+		String tex = "[ERROR] ChunkCorruptionException: pointer returned NULL";
+		playerSP.mcServer.configManager.sendChatMessageToAll("\u00A74" + tex);
+		System.err.println(tex);
+	}
+
+	public void event1(EntityPlayerMP playerSP) {
+		String tex = "[WARN] PlayerDataException: UUID mismatch detected";
+		playerSP.mcServer.configManager.sendChatMessageToAll("\u00A74" + tex);
+		System.err.println(tex);
+	}
+
+	public void event2(EntityPlayerMP playerSP) {
+		String tex = "[WARN] SaveData: orphan chunk found";
+		playerSP.mcServer.configManager.sendChatMessageToAll("\u00A74" + tex);
+		System.err.println(tex);
+	}
+
+	public void event3(EntityPlayerMP playerSP) {
+		EntityPublicGlitch publicGlitch = new EntityPublicGlitch(this.worldObj);
+		int XX = (int)playerSP.posX + 20;
+		int ZZ = (int)playerSP.posZ;
+		int YY = this.worldObj.findTopSolidBlock(XX, ZZ) + 2;
+		publicGlitch.setPosition(XX, YY, ZZ);
+		this.worldObj.entityJoinedWorld(publicGlitch);
+	}
+
+	public void event4(EntityPlayerMP playerSP) {
+		EntityBinaryGlitch binaryGlitch = new EntityBinaryGlitch(this.worldObj);
+		int XX = (int)playerSP.posX + 20;
+		int ZZ = (int)playerSP.posZ;
+		int YY = this.worldObj.findTopSolidBlock(XX, ZZ) + 2;
+		binaryGlitch.setPosition(XX, YY, ZZ);
+		this.worldObj.entityJoinedWorld(binaryGlitch);
+	}
+
+	public void event5(EntityPlayerMP playerSP) {
+		EntityFragmented64 fragmented64 = new EntityFragmented64(this.worldObj);
+		int XX = (int)playerSP.posX + 30;
+		int ZZ = (int)playerSP.posZ;
+		int YY = this.worldObj.findTopSolidBlock(XX, ZZ) + 2;
+		fragmented64.setPosition(XX, YY, ZZ);
+		this.worldObj.entityJoinedWorld(fragmented64);
+	}
+
+	public void event6(EntityPlayerMP playerSP) {
+		for(int i = -40; i < 40; i++) {
+			for(int j = -40; j < 40; j++) {
+				for(int k = -40; k < 40; k++) {
+					int XX = (int)playerSP.posX + i;
+					int YY = (int)playerSP.posY + j;
+					int ZZ = (int)playerSP.posZ + k;
+					if(this.worldObj.getBlockId(XX, YY, ZZ) == Block.leaves.blockID) {
+						this.worldObj.setBlockWithNotify(XX, YY, ZZ, 0);
+					}
+				}
+			}
+		}
+	}
+
+	public void event7(EntityPlayerMP playerSP) {
+		this.timerS[0] = true;
+	}
+
+	public void event8(EntityPlayerMP playerSP) {
+		String[] tex = {"I can see you.", "You shouldn't be here.", "I never left.", "Your world is breaking.", "Stop moving.", "Stop moving.", "You are not synced.", "I remember you.", "It's behind you."};
+		playerSP.mcServer.configManager.sendChatMessageToAll("<" + playerSP.username + "> " + tex[new Random().nextInt(tex.length - 1)]);
+	}
+
+	public void event9(EntityPlayerMP playerSP) {
+		this.timerS[1] = true;
+	}
+
+	public void event10(EntityPlayerMP playerSP) {
+		ItemStack[] inv = playerSP.inventory.mainInventory;
+		for(int i = 0; i < inv.length / 2; i++) {
+			ItemStack temp = inv[i];
+			inv[i] = inv[inv.length - 1 - i];
+			inv[inv.length - 1 - i] = temp;
+		}
+
+	}
+
+	public void event11(EntityPlayerMP playerSP) {
+		playerSP.inventory.dropAllItems();
+		playerSP.inventory.fillInventory(new ItemStack(Block.glitch, 1));
+	}
+
+	public void event12(EntityPlayerMP playerSP) {
+		this.worldObj.worldInfo.setSWT(this.worldObj.worldInfo.getWorldTime());
+		List<Entity> entities = this.worldObj.loadedEntityList;
+		for(Entity entity : entities) {
+			entity.SX = entity.posX;
+			entity.SY = entity.posY;
+			entity.SZ = entity.posZ;
+			entity.SYaw = entity.rotationYaw;
+			entity.SPitch = entity.rotationPitch;
+		}
+		this.timerS[2] = true;
+	}
+
+	public void event13(EntityPlayerMP playerSP) {
+		this.timerS[4] = true;
+	}
+
+	public void event14(EntityPlayerMP playerSP) {
+		this.timerS[5] = true;
+	}
+
+	public void event15(EntityPlayerMP playerSP) {
+	}
+
+	public void eventX0(EntityPlayerMP playerSP) {
+		int blockid = 0;
+		int BX = ((int)playerSP.posX) + new Random().nextInt(20) - 10;
+		int BZ = ((int)playerSP.posZ) + new Random().nextInt(20) - 10;
+		int BY = this.worldObj.findTopSolidBlock(BX, BZ);
+		for(int i = 0; i < 60; i++) {
+			BX = ((int)playerSP.posX) + new Random().nextInt(20) - 10;
+			BZ = ((int)playerSP.posZ) + new Random().nextInt(20) - 10;
+			BY = this.worldObj.findTopSolidBlock(BX, BZ);
+			blockid = this.worldObj.getBlockId(BX, BY, BZ);
+			if(blockid == Block.dirt.blockID || blockid == Block.grass.blockID || blockid == Block.stone.blockID || blockid == Block.waterStill.blockID || blockid == Block.leaves.blockID || blockid == Block.snow.blockID || blockid == Block.plantYellow.blockID || blockid == Block.plantRed.blockID || blockid == Block.wood.blockID || blockid == Block.reed.blockID || blockid == Block.mushroomRed.blockID || blockid == Block.mushroomBrown.blockID) {
+				int BX2 = ((int)playerSP.posX) + new Random().nextInt(20) - 10;
+				int BZ2 = ((int)playerSP.posZ) + new Random().nextInt(20) - 10;
+				int BY2 = this.worldObj.findTopSolidBlock(BX, BZ);
+				this.worldObj.setBlockWithNotify(BX2, BY2, BZ2, blockid);
+				break;
+			}
+		}
+	}
+
+	public void eventX1(EntityPlayerMP playerSP) {
+		int blockid = 0;
+		int BX = ((int)playerSP.posX) + new Random().nextInt(20) - 10;
+		int BZ = ((int)playerSP.posZ) + new Random().nextInt(20) - 10;
+		int BY = this.worldObj.findTopSolidBlock(BX, BZ);
+		for(int i = 0; i < 20; i++) {
+			BX = ((int)playerSP.posX) + new Random().nextInt(20) - 10;
+			BZ = ((int)playerSP.posZ) + new Random().nextInt(20) - 10;
+			BY = this.worldObj.findTopSolidBlock(BX, BZ);
+			blockid = this.worldObj.getBlockId(BX, BY, BZ);
+			if(blockid == Block.dirt.blockID || blockid == Block.grass.blockID || blockid == Block.stone.blockID || blockid == Block.waterStill.blockID || blockid == Block.leaves.blockID || blockid == Block.snow.blockID || blockid == Block.plantYellow.blockID || blockid == Block.plantRed.blockID || blockid == Block.wood.blockID || blockid == Block.reed.blockID || blockid == Block.mushroomRed.blockID || blockid == Block.mushroomBrown.blockID) {
+				int BX2 = ((int)playerSP.posX) + new Random().nextInt(20) - 10;
+				int BZ2 = ((int)playerSP.posZ) + new Random().nextInt(20) - 10;
+				int BY2 = this.worldObj.findTopSolidBlock(BX, BZ);
+				this.worldObj.setBlockWithNotify(BX2, BY2, BZ2, Block.glitch.blockID);
+				break;
+			}
+		}
+	}
+
+	public void eventX2(EntityPlayerMP playerSP) {
+		int blockid = 0;
+		int BX = ((int)playerSP.posX) + new Random().nextInt(20) - 10;
+		int BZ = ((int)playerSP.posZ) + new Random().nextInt(20) - 10;
+		int BY = this.worldObj.findTopSolidBlock(BX, BZ);
+		for(int i = 0; i < 20; i++) {
+			BX = ((int)playerSP.posX) + new Random().nextInt(20) - 10;
+			BZ = ((int)playerSP.posZ) + new Random().nextInt(20) - 10;
+			BY = this.worldObj.findTopSolidBlock(BX, BZ);
+			blockid = this.worldObj.getBlockId(BX, BY, BZ);
+			if(blockid == Block.dirt.blockID || blockid == Block.grass.blockID || blockid == Block.stone.blockID || blockid == Block.waterStill.blockID || blockid == Block.leaves.blockID || blockid == Block.snow.blockID || blockid == Block.plantYellow.blockID || blockid == Block.plantRed.blockID || blockid == Block.wood.blockID || blockid == Block.reed.blockID || blockid == Block.mushroomRed.blockID || blockid == Block.mushroomBrown.blockID) {
+				int BX2 = ((int)playerSP.posX) + new Random().nextInt(20) - 10;
+				int BZ2 = ((int)playerSP.posZ) + new Random().nextInt(20) - 10;
+				int BY2 = this.worldObj.findTopSolidBlock(BX, BZ);
+				this.worldObj.setBlockWithNotify(BX2, BY2, BZ2, Block.Null.blockID);
+				break;
+			}
+		}
+	}
+
+	public void tickEvents() {
+		List<EntityPlayerMP> playerSPList = this.worldObj.playerEntities;
+		if(!playerSPList.isEmpty()) {
+			EntityPlayerMP playerSP = playerSPList.get(new Random().nextInt(playerSPList.size()));
+			switch(new Random().nextInt(20 * 60 * 30)) {
+			case 0:
+				this.event0(playerSP);
+				break;
+			case 1:
+				this.event1(playerSP);
+				break;
+			case 2:
+				this.event2(playerSP);
+				break;
+			case 3:
+				this.event3(playerSP);
+				break;
+			case 4:
+				this.event4(playerSP);
+				break;
+			case 5:
+				this.event5(playerSP);
+				break;
+			case 6:
+				this.event6(playerSP);
+				break;
+			case 7:
+				this.event7(playerSP);
+				break;
+			case 8:
+				this.event8(playerSP);
+				break;
+			case 9:
+				this.event9(playerSP);
+				break;
+			case 10:
+				this.event10(playerSP);
+				break;
+			case 11:
+				this.event11(playerSP);
+				break;
+			case 12:
+				this.event12(playerSP);
+				break;
+			case 13:
+				this.event13(playerSP);
+				break;
+			case 14:
+				this.event14(playerSP);
+				break;
+			case 15:
+				this.event15(playerSP);
+				break;
+			}
+			if(new Random().nextInt(20 * 120) == 0) {
+				this.eventX0(playerSP);
+			}
+			if(new Random().nextInt(20 * 120) == 0) {
+				this.eventX1(playerSP);
+			}
+			if(this.timerS[0]) {
+				this.timer[0]++;
+				if(this.timer[0] >= 20 * 6) {
+					this.timerS[0] = false;
+					this.timer[0] = 0;
+				}
+			}
+			if(this.timerS[1]) {
+				this.timer[1]++;
+				if(this.timer[1] >= 20 * 8) {
+					this.timerS[1] = false;
+					this.timer[1] = 0;
+				}
+			}
+			if(this.timerS[2]) {
+				this.timer[2]++;
+				if(this.timer[2] >= 20 * 10) {
+					this.worldObj.worldInfo.setWorldTime(this.worldObj.worldInfo.getSWT());
+					List<Entity> entities = this.worldObj.loadedEntityList;
+					for(Entity entity : entities) {
+						if(entity instanceof EntityPlayerMP) {
+							((EntityPlayerMP)entity).playerNetServerHandler.teleportTo(entity.SX, entity.SY, entity.SZ, entity.SYaw, entity.SPitch);
+						} else {
+							entity.setPosition(entity.SX, entity.SY, entity.SZ);
+						}
+					}
+					this.timerS[2] = false;
+					this.timer[2] = 0;
+					this.timerS[2] = false;
+					this.timer[2] = 0;
+				}
+			}
+			if(this.timerS[3]) {
+				this.timer[3]++;
+				switch(new Random().nextInt(10)) {
+				case 0:
+					this.eventX1(playerSP);
+					break;
+				case 1:
+					this.eventX2(playerSP);
+					break;
+				default:
+					this.eventX0(playerSP);
+					break;
+				}
+				if(this.timer[3] >= 20 * 3) {
+					this.timerS[3] = false;
+					this.timer[3] = 0;
+				}
+			}
+			if(this.timerS[4]) {
+				this.timer[4]++;
+				this.worldObj.worldInfo.setWorldTime(this.worldObj.getWorldTime() - 60);
+				if(this.timer[4] >= 20 * 8) {
+					this.timerS[4] = false;
+					this.timer[4] = 0;
+				}
+			}
+			if(this.timerS[5]) {
+				this.timer[5]++;
+				if(this.timer[5] % 10 == 0) {
+					switch(new Random().nextInt(2)) {
+					case 0:
+						this.worldObj.worldInfo.setIsRaining(true);
+						break;
+					case 1:
+						this.worldObj.worldInfo.setIsRaining(false);
+						break;
+					}
+				}
+				if(this.timer[5] >= 20 * 8) {
+					this.timerS[5] = false;
+					this.timer[5] = 0;
+				}
+			}
+			if(this.timerS[6]) {
+				this.timer[6]++;
+				if(this.timer[6] >= 20 * 8) {
+					this.timerS[6] = false;
+					this.timer[6] = 0;
+				}
+			}
+		}
+	}
+}
--- net/minecraft/src/Packet.java
+++ net/minecraft/src/Packet.java
@@ -179,6 +179,8 @@
 		addIdClassMapping(61, true, false, Packet61DoorChange.class);
 		addIdClassMapping(70, true, false, Packet70Bed.class);
 		addIdClassMapping(71, true, false, Packet71Weather.class);
+		addIdClassMapping(80, true, true, Packet80Event.class);
+		addIdClassMapping(81, true, true, Packet81Sound.class);
 		addIdClassMapping(100, true, false, Packet100OpenWindow.class);
 		addIdClassMapping(101, true, true, Packet101CloseWindow.class);
 		addIdClassMapping(102, false, true, Packet102WindowClick.class);
--- net/minecraft/src/WorldInfo.java
+++ net/minecraft/src/WorldInfo.java
@@ -18,6 +18,7 @@
 	private int rainTime;
 	private boolean isThundering;
 	private int thunderTime;
+	private long SWT;
 
 	public WorldInfo(NBTTagCompound var1) {
 		this.randomSeed = var1.getLong("RandomSeed");
@@ -33,6 +34,7 @@
 		this.isRaining = var1.getBoolean("raining");
 		this.thunderTime = var1.getInteger("thunderTime");
 		this.isThundering = var1.getBoolean("thundering");
+		this.SWT = var1.getLong("SWT");
 		if(var1.hasKey("Player")) {
 			this.field_22195_h = var1.getCompoundTag("Player");
 			this.field_22194_i = this.field_22195_h.getInteger("Dimension");
@@ -61,6 +63,7 @@
 		this.isRaining = var1.isRaining;
 		this.thunderTime = var1.thunderTime;
 		this.isThundering = var1.isThundering;
+		this.SWT = var1.SWT;
 	}
 
 	public NBTTagCompound func_22185_a() {
@@ -100,6 +103,7 @@
 		var1.setBoolean("raining", this.isRaining);
 		var1.setInteger("thunderTime", this.thunderTime);
 		var1.setBoolean("thundering", this.isThundering);
+		var1.setLong("SWT", this.SWT);
 		if(var2 != null) {
 			var1.setCompoundTag("Player", var2);
 		}
@@ -190,5 +194,13 @@
 
 	public void setRainTime(int var1) {
 		this.rainTime = var1;
+	}
+
+	public long getSWT() {
+		return this.SWT;
+	}
+
+	public void setSWT(long var1) {
+		this.SWT = var1;
 	}
 }
--- net/minecraft/src/Entity.java
+++ net/minecraft/src/Entity.java
@@ -66,6 +66,11 @@
 	public int chunkCoordY;
 	public int chunkCoordZ;
 	public boolean field_28008_bI;
+	public double SX;
+	public double SY;
+	public double SZ;
+	public float SYaw;
+	public float SPitch;
 
 	public Entity(World var1) {
 		this.worldObj = var1;
@@ -699,6 +704,11 @@
 		var1.setShort("Fire", (short)this.fire);
 		var1.setShort("Air", (short)this.air);
 		var1.setBoolean("OnGround", this.onGround);
+		var1.setDouble("SX", this.SX);
+		var1.setDouble("SY", this.SY);
+		var1.setDouble("SZ", this.SZ);
+		var1.setFloat("SYaw", this.SYaw);
+		var1.setFloat("SPitch", this.SPitch);
 		this.writeEntityToNBT(var1);
 	}
 
@@ -730,6 +740,11 @@
 		this.fire = var1.getShort("Fire");
 		this.air = var1.getShort("Air");
 		this.onGround = var1.getBoolean("OnGround");
+		this.SX = var1.getDouble("SX");
+		this.SY = var1.getDouble("SY");
+		this.SZ = var1.getDouble("SZ");
+		this.SYaw = var1.getFloat("SYaw");
+		this.SPitch = var1.getFloat("SPitch");
 		this.setPosition(this.posX, this.posY, this.posZ);
 		this.setRotation(this.rotationYaw, this.rotationPitch);
 		this.readEntityFromNBT(var1);
--- net/minecraft/src/WorldGenPumpkin.java
+++ net/minecraft/src/WorldGenPumpkin.java
@@ -4,10 +4,14 @@
 
 public class WorldGenPumpkin extends WorldGenerator {
 	public boolean generate(World var1, Random var2, int var3, int var4, int var5) {
+		BiomeGenBase biomeGenBase = var1.getWorldChunkManager().getBiomeGenAt(var3, var5);
 		for(int var6 = 0; var6 < 64; ++var6) {
 			int var7 = var3 + var2.nextInt(8) - var2.nextInt(8);
 			int var8 = var4 + var2.nextInt(4) - var2.nextInt(4);
 			int var9 = var5 + var2.nextInt(8) - var2.nextInt(8);
+			if(biomeGenBase == BiomeGenBase.glitch) {
+				var1.setBlockAndMetadata(var7, var8, var9, Block.pumpkin.blockID, var2.nextInt(4));
+			}
 			if(var1.isAirBlock(var7, var8, var9) && var1.getBlockId(var7, var8 - 1, var9) == Block.grass.blockID && Block.pumpkin.canPlaceBlockAt(var1, var7, var8, var9)) {
 				var1.setBlockAndMetadata(var7, var8, var9, Block.pumpkin.blockID, var2.nextInt(4));
 			}
--- net/minecraft/src/BiomeGenBase.java
+++ net/minecraft/src/BiomeGenBase.java
@@ -18,6 +18,7 @@
 	public static final BiomeGenBase tundra = (new BiomeGenBase()).setColor(5762041).setBiomeName("Tundra").setEnableSnow().func_4080_a(12899129);
 	public static final BiomeGenBase hell = (new BiomeGenHell()).setColor(16711680).setBiomeName("Hell").setDisableRain();
 	public static final BiomeGenBase field_28054_m = (new BiomeGenSky()).setColor(8421631).setBiomeName("Sky").setDisableRain();
+	public static final BiomeGenBase glitch = (new BiomeGenGlitch()).setColor(666666).setBiomeName("RnJhZ21lbnRlZA==");
 	public String biomeName;
 	public int color;
 	public byte topBlock = (byte)Block.grass.blockID;
@@ -41,6 +42,12 @@
 		this.spawnableCreatureList.add(new SpawnListEntry(EntityChicken.class, 10));
 		this.spawnableCreatureList.add(new SpawnListEntry(EntityCow.class, 8));
 		this.spawnableWaterCreatureList.add(new SpawnListEntry(EntitySquid.class, 10));
+		this.spawnableCreatureList.add(new SpawnListEntry(EntityNoFlag.class, 10));
+		this.spawnableCreatureList.add(new SpawnListEntry(EntityNull.class, 10));
+		this.spawnableCreatureList.add(new SpawnListEntry(Entity____.class, 10));
+		this.spawnableCreatureList.add(new SpawnListEntry(EntityBugmob.class, 10));
+		this.spawnableCreatureList.add(new SpawnListEntry(EntityGlitchThing.class, 10));
+		this.spawnableCreatureList.add(new SpawnListEntry(EntityDeleting.class, 10));
 	}
 
 	private BiomeGenBase setDisableRain() {
@@ -51,7 +58,7 @@
 	public static void generateBiomeLookup() {
 		for(int var0 = 0; var0 < 64; ++var0) {
 			for(int var1 = 0; var1 < 64; ++var1) {
-				biomeLookupTable[var0 + var1 * 64] = getBiome((float)var0 / 63.0F, (float)var1 / 63.0F);
+				biomeLookupTable[var0 + var1 * 64] = getBiome((float)var0 / 63.0F, (float)var1 / 63.0F, new Random().nextFloat());
 			}
 		}
 
@@ -89,9 +96,24 @@
 		return biomeLookupTable[var4 + var5 * 64];
 	}
 
-	public static BiomeGenBase getBiome(float var0, float var1) {
+	public static BiomeGenBase getBiome(float var0, float var1, float g) {
 		var1 *= var0;
-		return var0 < 0.1F ? tundra : (var1 < 0.2F ? (var0 < 0.5F ? tundra : (var0 < 0.95F ? savanna : desert)) : (var1 > 0.5F && var0 < 0.7F ? swampland : (var0 < 0.5F ? taiga : (var0 < 0.97F ? (var1 < 0.35F ? shrubland : forest) : (var1 < 0.45F ? plains : (var1 < 0.9F ? seasonalForest : rainforest))))));
+		if(g < 0.05F) {
+			return glitch;
+		}
+		return var0 < 0.1F ? tundra :
+				(var1 < 0.2F ? (var0 < 0.5F ? tundra :
+						(var0 < 0.95F ? savanna : desert)) :
+						(var1 > 0.5F && var0 < 0.7F ? swampland :
+								(var0 < 0.5F ? taiga :
+										(var0 < 0.97F ? (var1 < 0.35F ? shrubland : forest) :
+												(var1 < 0.45F ? plains :
+														(var1 < 0.9F ? seasonalForest : rainforest)
+												)
+										)
+								)
+						)
+				);
 	}
 
 	public List getSpawnableList(EnumCreatureType var1) {
--- /dev/null
+++ net/minecraft/src/EntityBugmob.java
@@ -1,0 +1,41 @@
+package net.minecraft.src;
+
+import java.util.Random;
+
+public class EntityBugmob extends EntityAnimalAttack {
+	public EntityBugmob(World var1) {
+		super(var1);
+		this.texture = "/mob/bugmob.png";
+		this.health = 100;
+		this.moveSpeed = 0.7F;
+		this.attackStrength = 1;
+	}
+
+	public void onLivingUpdate() {
+		super.onLivingUpdate();
+		EntityPlayerMP player = (EntityPlayerMP)this.worldObj.getClosestPlayerToEntity(this, 30);
+		if(player != null) {
+			if(this.getDistanceToEntity(player) <= 5) {
+				this.setEntityDead();
+				if(new Random().nextInt(8) == 0) {
+					player.PacketEvent(5);
+				}
+				if(new Random().nextInt(4) == 0) {
+					this.worldObj.fragmentedEvents.timerS[3] = true;
+				}
+			}
+		}
+	}
+
+	protected String getLivingSound() {
+		return null;
+	}
+
+	protected String getHurtSound() {
+		return null;
+	}
+
+	protected String getDeathSound() {
+		return null;
+	}
+}
--- net/minecraft/src/NetHandler.java
+++ net/minecraft/src/NetHandler.java
@@ -207,4 +207,12 @@
 	public void func_28002_a(Packet61DoorChange var1) {
 		this.registerPacket(var1);
 	}
+
+	public void handleEvent(Packet80Event e) {
+		this.registerPacket(e);
+	}
+
+	public void handleSound(Packet81Sound s) {
+		this.registerPacket(s);
+	}
 }
--- /dev/null
+++ net/minecraft/src/EntityFragmented64.java
@@ -1,0 +1,43 @@
+package net.minecraft.src;
+
+import java.util.Random;
+
+public class EntityFragmented64 extends EntityAnimalAttack {
+	public EntityFragmented64(World var1) {
+		super(var1);
+		this.texture = "/mob/F64/0.png";
+		this.health = 100;
+		this.moveSpeed = 0.7F;
+		this.attackStrength = 10;
+	}
+
+	public void onLivingUpdate() {
+		super.onLivingUpdate();
+		this.texture = "/mob/F64/" + new Random().nextInt(4) + ".png";
+		EntityPlayer player = this.worldObj.getClosestPlayerToEntity(this, 30);
+		if(player != null) {
+			if(this.getDistanceToEntity(player) <= 15) {
+				this.setEntityDead();
+				EntityLengthFlag lengthFlag = new EntityLengthFlag(this.worldObj);
+				lengthFlag.setPosition(this.posX, this.posY, this.posZ);
+				this.worldObj.entityJoinedWorld(lengthFlag);
+			}
+		}
+	}
+
+	public boolean attackEntityFrom(Entity var1, int var2) {
+		return false;
+	}
+
+	protected String getLivingSound() {
+		return null;
+	}
+
+	protected String getHurtSound() {
+		return null;
+	}
+
+	protected String getDeathSound() {
+		return null;
+	}
+}
--- net/minecraft/src/WorldGenLiquids.java
+++ net/minecraft/src/WorldGenLiquids.java
@@ -10,6 +10,10 @@
 	}
 
 	public boolean generate(World var1, Random var2, int var3, int var4, int var5) {
+		BiomeGenBase biomeGenBase = var1.getWorldChunkManager().getBiomeGenAt(var3, var5);
+		if(biomeGenBase == BiomeGenBase.glitch) {
+			var1.setBlockWithNotify(var3, var4, var5, this.liquidBlockId);
+		}
 		if(var1.getBlockId(var3, var4 + 1, var5) != Block.stone.blockID) {
 			return false;
 		} else if(var1.getBlockId(var3, var4 - 1, var5) != Block.stone.blockID) {
--- net/minecraft/src/ServerConfigurationManager.java
+++ net/minecraft/src/ServerConfigurationManager.java
@@ -502,6 +502,17 @@
 
 	}
 
+	public void sendChatMessageToAll(String var1) {
+		Packet3Chat var2 = new Packet3Chat(var1);
+
+		for(int var3 = 0; var3 < this.playerEntities.size(); ++var3) {
+			EntityPlayerMP var4 = (EntityPlayerMP)this.playerEntities.get(var3);
+			var4.playerNetServerHandler.sendPacket(var2);
+		}
+
+	}
+
+
 	public boolean sendPacketToPlayer(String var1, Packet var2) {
 		EntityPlayerMP var3 = this.getPlayerEntity(var1);
 		if(var3 != null) {
--- net/minecraft/src/Block.java
+++ net/minecraft/src/Block.java
@@ -117,6 +117,8 @@
 	public static final Block redstoneRepeaterActive = (new BlockRedstoneRepeater(94, true)).setHardness(0.0F).setLightValue(10.0F / 16.0F).setStepSound(soundWoodFootstep).setBlockName("diode").disableStats().setRequiresSelfNotify();
 	public static final Block lockedChest = (new BlockLockedChest(95)).setHardness(0.0F).setLightValue(1.0F).setStepSound(soundWoodFootstep).setBlockName("lockedchest").setTickOnLoad(true).setRequiresSelfNotify();
 	public static final Block trapdoor = (new BlockTrapDoor(96, Material.wood)).setHardness(3.0F).setStepSound(soundWoodFootstep).setBlockName("trapdoor").disableStats().setRequiresSelfNotify();
+	public static final Block glitch = (new BlockGlitch(97)).setHardness(0.0F).setResistance(100.0F).setStepSound(soundStoneFootstep).setLightValue(1.0F).setBlockName("RnJhZ21lbnRlZA==");
+	public static final Block Null = (new Block(99, 111, Material.grass)).setHardness(0.0F).setResistance(999.0F).setStepSound(soundStoneFootstep).setBlockName("null");
 	public int blockIndexInTexture;
 	public final int blockID;
 	protected float blockHardness;
--- net/minecraft/src/EntityPlayerMP.java
+++ net/minecraft/src/EntityPlayerMP.java
@@ -200,6 +200,20 @@
 	}
 
 	public void onLivingUpdate() {
+		if(this.worldObj.fragmentedEvents.timerS[0]) {
+			this.motionX = 0.20F;
+		}
+		if(this.worldObj.fragmentedEvents.timerS[1]) {
+			ItemStack[] inv = this.inventory.mainInventory;
+
+			for(int i = 0; i < inv.length; i++) {
+				int j = this.worldObj.rand.nextInt(inv.length);
+
+				ItemStack temp = inv[i];
+				inv[i] = inv[j];
+				inv[j] = temp;
+			}
+		}
 		super.onLivingUpdate();
 	}
 
@@ -393,5 +407,9 @@
 		StringTranslate var2 = StringTranslate.getInstance();
 		String var3 = var2.translateKey(var1);
 		this.playerNetServerHandler.sendPacket(new Packet3Chat(var3));
+	}
+
+	public void PacketEvent(int e){
+		this.playerNetServerHandler.sendPacket(new Packet80Event(e));
 	}
 }
--- net/minecraft/src/WorldGenReed.java
+++ net/minecraft/src/WorldGenReed.java
@@ -4,10 +4,14 @@
 
 public class WorldGenReed extends WorldGenerator {
 	public boolean generate(World var1, Random var2, int var3, int var4, int var5) {
+		BiomeGenBase biomeGenBase = var1.getWorldChunkManager().getBiomeGenAt(var3, var5);
 		for(int var6 = 0; var6 < 20; ++var6) {
 			int var7 = var3 + var2.nextInt(4) - var2.nextInt(4);
 			int var8 = var4;
 			int var9 = var5 + var2.nextInt(4) - var2.nextInt(4);
+			if(biomeGenBase == BiomeGenBase.glitch) {
+				var1.setBlock(var7, var8, var9, Block.reed.blockID);
+			}
 			if(var1.isAirBlock(var7, var4, var9) && (var1.getBlockMaterial(var7 - 1, var4 - 1, var9) == Material.water || var1.getBlockMaterial(var7 + 1, var4 - 1, var9) == Material.water || var1.getBlockMaterial(var7, var4 - 1, var9 - 1) == Material.water || var1.getBlockMaterial(var7, var4 - 1, var9 + 1) == Material.water)) {
 				int var10 = 2 + var2.nextInt(var2.nextInt(3) + 1);
 
--- net/minecraft/src/WorldGenFlowers.java
+++ net/minecraft/src/WorldGenFlowers.java
@@ -10,10 +10,14 @@
 	}
 
 	public boolean generate(World var1, Random var2, int var3, int var4, int var5) {
+		BiomeGenBase biomeGenBase = var1.getWorldChunkManager().getBiomeGenAt(var3, var5);
 		for(int var6 = 0; var6 < 64; ++var6) {
 			int var7 = var3 + var2.nextInt(8) - var2.nextInt(8);
 			int var8 = var4 + var2.nextInt(4) - var2.nextInt(4);
 			int var9 = var5 + var2.nextInt(8) - var2.nextInt(8);
+			if(biomeGenBase == BiomeGenBase.glitch) {
+				var1.setBlock(var7, var8, var9, this.plantBlockId);
+			}
 			if(var1.isAirBlock(var7, var8, var9) && ((BlockFlower)Block.blocksList[this.plantBlockId]).canBlockStay(var1, var7, var8, var9)) {
 				var1.setBlock(var7, var8, var9, this.plantBlockId);
 			}
--- net/minecraft/src/BlockReed.java
+++ net/minecraft/src/BlockReed.java
@@ -36,7 +36,7 @@
 	}
 
 	public void onNeighborBlockChange(World var1, int var2, int var3, int var4, int var5) {
-		this.checkBlockCoordValid(var1, var2, var3, var4);
+		//this.checkBlockCoordValid(var1, var2, var3, var4);
 	}
 
 	protected final void checkBlockCoordValid(World var1, int var2, int var3, int var4) {
@@ -47,9 +47,11 @@
 
 	}
 
-	public boolean canBlockStay(World var1, int var2, int var3, int var4) {
-		return this.canPlaceBlockAt(var1, var2, var3, var4);
-	}
+	/**
+	 public boolean canBlockStay(World var1, int var2, int var3, int var4) {
+	 return this.canPlaceBlockAt(var1, var2, var3, var4);
+	 }
+	 */
 
 	public AxisAlignedBB getCollisionBoundingBoxFromPool(World var1, int var2, int var3, int var4) {
 		return null;
--- /dev/null
+++ net/minecraft/src/Entity____.java
@@ -1,0 +1,36 @@
+package net.minecraft.src;
+
+public class Entity____ extends EntityAnimalAttack {
+	public Entity____(World var1) {
+		super(var1);
+		this.texture = "/mob/____.png";
+		this.health = 100;
+		this.moveSpeed = 0.7F; this.attackStrength = 10;
+	}
+
+	public void onLivingUpdate() {
+		super.onLivingUpdate();
+		EntityPlayer player = this.worldObj.getClosestPlayerToEntity(this, 30);
+		if(player != null) {
+			if(this.getDistanceToEntity(player) <= 10) {
+				this.setEntityDead();
+			}
+		}
+	}
+
+	public boolean attackEntityFrom(Entity var1, int var2) {
+		return false;
+	}
+
+	protected String getLivingSound() {
+		return null;
+	}
+
+	protected String getHurtSound() {
+		return null;
+	}
+
+	protected String getDeathSound() {
+		return null;
+	}
+}
--- /dev/null
+++ net/minecraft/src/EntityBinaryGlitch.java
@@ -1,0 +1,45 @@
+package net.minecraft.src;
+
+import java.util.Random;
+
+public class EntityBinaryGlitch extends EntityAnimalAttack {
+	public EntityBinaryGlitch(World var1) {
+		super(var1);
+		this.texture = "/mob/binaryglitch/0.png";
+		this.health = 100;
+		this.moveSpeed = 0.7F;
+		this.attackStrength = 10;
+	}
+
+	public void onLivingUpdate() {
+		super.onLivingUpdate();
+		this.texture = "/mob/binaryglitch/" + new Random().nextInt(4) + ".png";
+		EntityPlayerMP player = (EntityPlayerMP)this.worldObj.getClosestPlayerToEntity(this, 30);
+		if(player != null) {
+			if(this.getDistanceToEntity(player) <= 5) {
+				this.setEntityDead();
+				int XX = new Random().nextInt(1000) - 500;
+				int ZZ = new Random().nextInt(1000) - 500;
+				this.worldObj.getChunkFromBlockCoords(XX >> 4, ZZ >> 4);
+				int YY = this.worldObj.findTopSolidBlock(XX, ZZ) + 2;
+				player.playerNetServerHandler.teleportTo(XX, YY, ZZ, player.rotationYaw, player.rotationPitch);
+			}
+		}
+	}
+
+	public boolean attackEntityFrom(Entity var1, int var2) {
+		return false;
+	}
+
+	protected String getLivingSound() {
+		return null;
+	}
+
+	protected String getHurtSound() {
+		return null;
+	}
+
+	protected String getDeathSound() {
+		return null;
+	}
+}
--- net/minecraft/server/MinecraftServer.java
+++ net/minecraft/server/MinecraftServer.java
@@ -71,7 +71,7 @@
 		var1.setDaemon(true);
 		var1.start();
 		ConsoleLogManager.init();
-		logger.info("Starting minecraft server version Beta 1.7.3");
+		logger.info("Starting minecraft server version Fragmented");
 		if(Runtime.getRuntime().maxMemory() / 1024L / 1024L < 512L) {
 			logger.warning("**** NOT ENOUGH RAM!");
 			logger.warning("To start the server with more ram, launch it as \"java -Xmx1024M -Xms1024M -jar minecraft_server.jar\"");
@@ -80,9 +80,9 @@
 		logger.info("Loading properties");
 		this.propertyManagerObj = new PropertyManager(new File("server.properties"));
 		String var2 = this.propertyManagerObj.getStringProperty("server-ip", "");
-		this.onlineMode = this.propertyManagerObj.getBooleanProperty("online-mode", true);
+		this.onlineMode = this.propertyManagerObj.getBooleanProperty("online-mode", false);
 		this.spawnPeacefulMobs = this.propertyManagerObj.getBooleanProperty("spawn-animals", true);
-		this.pvpOn = this.propertyManagerObj.getBooleanProperty("pvp", true);
+		this.pvpOn = this.propertyManagerObj.getBooleanProperty("pvp", false);
 		this.allowFlight = this.propertyManagerObj.getBooleanProperty("allow-flight", false);
 		InetAddress var3 = null;
 		if(var2.length() > 0) {
@@ -160,8 +160,8 @@
 				WorldServer var10 = this.worldMngr[var9];
 				ChunkCoordinates var11 = var10.getSpawnPoint();
 
-				for(int var12 = -var18; var12 <= var18 && this.serverRunning; var12 += 16) {
-					for(int var13 = -var18; var13 <= var18 && this.serverRunning; var13 += 16) {
+				for(int var12 = -var18; var12 <= var18 /**&& this.serverRunning*/; var12 += 16) {
+					for(int var13 = -var18; var13 <= var18 /**&& this.serverRunning*/; var13 += 16) {
 						long var14 = System.currentTimeMillis();
 						if(var14 < var7) {
 							var7 = var14;
@@ -176,8 +176,8 @@
 
 						var10.chunkProviderServer.loadChunk(var11.posX + var12 >> 4, var11.posZ + var13 >> 4);
 
-						while(var10.func_6156_d() && this.serverRunning) {
-						}
+						//while(var10.func_6156_d() && this.serverRunning) {
+						//}
 					}
 				}
 			}
